File: core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/promisehandlers/AddOutputPolicyHandler.scala
package edu.uci.ics.amber.engine.architecture.worker.neo.promisehandlers

import edu.uci.ics.amber.engine.architecture.sendsemantics.datatransferpolicy.DataSendingPolicy
import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerAsyncRPCHandlerInitializer
import edu.uci.ics.amber.engine.architecture.worker.neo.promisehandlers.AddOutputPolicyHandler.AddOutputPolicy
import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.{CommandCompleted, ControlCommand}
import edu.uci.ics.amber.engine.common.statetransition.WorkerStateManager.Ready

object AddOutputPolicyHandler{
  final case class AddOutputPolicy(policy:DataSendingPolicy) extends ControlCommand[CommandCompleted]
}



trait AddOutputPolicyHandler {
  this: WorkerAsyncRPCHandlerInitializer =>

  registerHandler{
    (msg:AddOutputPolicy, sender) =>
      stateManager.confirmState(Ready)
      tupleToBatchConverter.addPolicy(msg.policy)
      CommandCompleted()
  }

}

File: core/amber/src/main/python/core/architecture/handlers/add_output_policy_handler.py
from proto.edu.uci.ics.amber.engine.architecture.worker import AddPartitioningV2
from .handler_base import Handler
from ..managers.context import Context


class AddOutputPolicyHandler(Handler):
    cmd = AddPartitioningV2

    def __call__(self, context: Context, command: AddPartitioningV2, *args, **kwargs):
        context.tuple_to_batch_converter.add_policy(command.partitioning)
        return None
